rust/src/logger.rs

```diff
@@ -54,21 +55,65 @@ where
         event: &tracing::Event<'_>,
         _ctx: tracing_subscriber::layer::Context<'_, S>,
     ) {
-        let metadata = event.metadata();
+        let mut fields = BTreeMap::new();
+        let mut visitor = JsonVisitor(&mut fields);
+        event.record(&mut visitor);
 
-        let msg = format!("{}: {}", metadata.name(), metadata.fields());
+        let fmt_fields = fields
+            .iter()
+            .map(|field| format!("{}: {}", field.0, field.1))
+            .collect::<Vec<String>>()
+            .join(",");
+        let msg = format!("{}: {}", event.metadata().name(), fmt_fields);
 
         LOG_STREAM_SINK
             .try_get()
             .expect("StreamSink from Flutter to be initialised")
             .add(LogEntry {
                 msg,
-                target: metadata.target().to_string(),
-                level: metadata.level().to_string(),
+                target: event.metadata().target().to_string(),
+                level: event.metadata().level().to_string(),
             });
     }
 }
 
+struct JsonVisitor<'a>(&'a mut BTreeMap<String, String>);
```

Comment:
nit
```suggestion
struct Visitor<'a>(&'a mut BTreeMap<String, String>);
```
(here and elsewhere - we don't use json anymore)